tests/framework/extensions/workloads/pods/pod_status.go

```diff
@@ -46,18 +46,30 @@ func StatusPods(client *rancher.Client, clusterID string) ([]string, []error) {
 				return false, err
 			}
 
+			if len(podStatus.ContainerStatuses) == 0 {
+				return false, nil
+			}
+
 			image := podStatus.ContainerStatuses[0].Image
 			phase := podStatus.Phase
+
 			if phase == corev1.PodFailed || phase == corev1.PodUnknown {
 				podErrors = append(podErrors, fmt.Errorf("ERROR: %s: %s", pod.Name, podStatus))
 				logrus.Infof("Pod %s: Not active | Image %s", pod.Name, image)
 				return false, nil
-			} else if phase == corev1.PodRunning {
```

Comment:
Why did you change the else if?

Reply:
Mainly personal preference; I find if there are a few conditions, it's more readable with multiple ifs rather than else ifs. I can change it, surely. Would you like me to change it back?

Reply:
I was trying to save checks but with the return it does that anyways. So ignore me lol